bonsai/Modules/BudgetDetails/BudgetViewModel.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  BudgetViewModel.swift
+//  bonsai
+//
+//  Created by antuan.khoanh on 17/07/2022.
+//
+
+import Foundation
+
+protocol BudgetViewModelProtocol {
+   var totalBudget: NSDecimalNumber { get }
+   var totalMoneyLeft: NSDecimalNumber { get }
+   var totalMoneySpent: NSDecimalNumber { get }
+   var budgetDaily: NSDecimalNumber { get }
+   var transactions: [Transaction] { get }
+}
+
+struct BudgetViewModel: BudgetViewModelProtocol {
+   
+   private let budgetService: BudgetCalculationServiceProtocol
+   public private(set) var transactions: [Transaction] = []
+   
+   init(budgetService: BudgetServiceProtocol, transactions: [Transaction]) {
+      self.budgetService = budgetService
+      self.transactions = transactions
+      /*
+       for manual test
+       1 delete app from simulator and uncomment createBudget()
+       2 create some transactions
+       */
+//      budgetService.createBudget(name: "Tailand", with: 9000, on: 30)
```

Comment:
Why does name not shown on the screen?

![Simulator Screen Shot - iPhone 13 - 2022-07-26 at 13 50 42](https://user-images.githubusercontent.com/38043698/180942184-29fabdec-13f2-4bcd-861f-b0d5b99c429d.png)

Reply:
updated+